cirq-core/cirq/protocols/kraus_protocol.py

```diff
@@ -162,6 +192,11 @@ def kraus(
     if default is not RaiseTypeErrorIfNotProvided:
         return default
 
+    # Last-resort fallback: try to derive Kraus from _apply_channel_
+    result = _strat_kraus_from_apply_channel(val)
+    if result is not None:
+        return result
```

Comment:
Please move before the line 192 with `if default is not RaiseTypeErrorIfNotProvided:`.

(In case I missed some discussion why this needs to be _after the default_, please explain here too.)

Reply:
thanks for catching!